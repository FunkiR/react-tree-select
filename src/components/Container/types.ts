import type {ComponentType, HTMLProps} from 'react';

export type Props = HTMLProps<HTMLDivElement>;

export type ContainerComponent = ComponentType<Props>;
