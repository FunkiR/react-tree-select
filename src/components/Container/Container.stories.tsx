import {Meta, Story} from '@storybook/react/types-6-0';
import React from 'react';

import Container from './Container';
import type {Props} from './types';

export default {
	title: 'Components/Container',
	component: Container
} as Meta;

const Template: Story<Props> = args => <Container {...args} />;

export const Default = Template.bind({});

Default.args = {
	children: 'test',
	title: 'test'
};

