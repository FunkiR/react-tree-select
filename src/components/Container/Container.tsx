import React, {forwardRef, memo} from 'react';

import type {ContainerComponent,Props} from "./types";

export const Container: ContainerComponent = forwardRef<HTMLDivElement, Props>((props, ref) => (
	<div {...props} ref={ref} />
));

export default memo(Container);
