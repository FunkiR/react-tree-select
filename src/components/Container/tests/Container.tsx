import Container from "components/Container";
import {shallow} from "enzyme";
import React from "react";

describe('Container component', () => {

	it('renders component', () => {
		const wrapper = shallow(<Container />);
		expect(wrapper).toMatchSnapshot();
	});

	it('has text rendered as a child', () => {
		const text = 'test';

		const wrapper = shallow(<Container>{text}</Container>);
		expect(wrapper.text()).toEqual(text);
	});

	it('has component rendered as a child', () => {
		const Component = () => <div />;

		const wrapper = shallow(<Container><Component /></Container>);
		expect(wrapper.exists(Component)).toBeTruthy();
	});

	it('has html attribute equal to title prop', () => {
		const title = 'title';

		const wrapper = shallow(<Container title={title} />);
		expect(wrapper.prop('title')).toEqual(title);
	});

	it('handles onClick properly and calls provided callback', () => {
		const mockOnClick = jest.fn();

		const wrapper = shallow(<Container onClick={mockOnClick} />);
		wrapper.find('div').simulate('click');

		expect(mockOnClick.mock.calls.length).toEqual(1);
	});
});

