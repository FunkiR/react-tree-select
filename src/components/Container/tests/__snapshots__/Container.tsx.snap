// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders component 1`] = `<div />`;

exports[`Container component renders correctly 1`] = `<div />`;
